core/plugins/org.fusesource.ide.camel.model.service.v2_17_0_redhat_630187/src/org/fusesource/ide/camel/model/service/internal/CamelService.java

```diff
@@ -34,18 +35,27 @@
 	
 	private CamelModelLoader loader;
 	private CamelCatalog catalog;
-	
+
 	/* (non-Javadoc)
 	 * @see org.fusesource.ide.camel.model.service.core.ICamelManagerService#getCamelModel()
 	 */
 	@Override
+	@Deprecated
 	public CamelModel getCamelModel() {
-		if (this.loader == null) this.loader = new XmlCamelModelLoader();
+		return getCamelModel(CamelModelFactory.RUNTIME_PROVIDER_KARAF);
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.fusesource.ide.camel.model.service.core.ICamelManagerService#getCamelModel(java.lang.String)
+	 */
+	@Override
+	public CamelModel getCamelModel(String runtimeProvider) {
+		this.loader = new XmlCamelModelLoader();
```

Comment:
the loader class attribute is now used only in this method, so attribute can be removed. Unless we want to keep a cache for it